lib/domain.gi

```diff
@@ -22,7 +22,11 @@ InstallMethod( \=,
     IsIdenticalObj,
     [ IsCollection and IsList, IsDomain ],
     function( C, D )
-    return IsSSortedList( C ) and AsSSortedList( D ) = C;
+    if HasIsFinite( D ) and not IsFinite( D ) then
+      return false;
```

Comment:
Actually... Why is this a correct? Collections can be infinite, and so can lists, no (at least I just couldn't find anything in the documentaton which excludes the latter, and the filter `IsList` does not imply `IsFinite`). Am I missing something?

Reply:
Indeed:

```
gap> rats := Enumerator(Rationals);
<enumerator of Rationals>
gap> IsList(rats)
true
gap> IsFinite(rats);
false
```

Reply:
So the test really needs to be more like this:
```gap
   if HasIsFinite(C) and HasIsFinite(D) and (IsFinite(C) <> IsFinite(D)) then
    return false;
  fi;
```
(I'd also not use the `fi` to keep the diff minimal)

Reply:
@fingolfin @markuspf pushed updated PR.